src/storage/src/store.rs

```diff
@@ -172,37 +227,64 @@ pub trait StateStore: Send + Sync + 'static + Clone {
     fn clear_shared_buffer(&self) -> Self::ClearSharedBufferFuture<'_> {
         todo!()
     }
+
+    fn new_local(&self, table_id: TableId) -> Self::NewLocalFuture<'_>;
 }
 
-pub trait StateStoreIter: Send + 'static {
-    type Item;
-    type NextFuture<'a>: Future<Output = StorageResult<Option<Self::Item>>> + Send;
+/// A state store that is dedicated for streaming operator, which only reads the uncommitted data
+/// written by itself. Each local state store is not `Clone`, and is owned by a streaming state
+/// table.
+pub trait LocalStateStore: StateStoreRead + StateStoreWrite + StaticSendSync {
+    /// Inserts a key-value entry associated with a given `epoch` into the state store.
+    fn insert(&self, _key: Bytes, _val: Bytes) -> StorageResult<()> {
+        unimplemented!()
+    }
```

Comment:
Can we remove these default implementations?

Reply:
These methods are not used yet, but we may use them later. If we remove the default implementation, we will have to add the `unimplemented` in all the types that implement `LocalStateStore`. So I will prefer keeping the default implementation.